### PR TITLE
CATROID-1120 Refactor BaseActivity and related components to Kotlin

### PR DESCRIPTION
**Pull request by khushal1707**
Ticket: https://jira.catrob.at/browse/CATROID-1120

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [ ] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [ ] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.kt [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            if (activeProcess == context.packageName) {
+                return false
+            }
+        }
+        return true
+    }
+
+    companion object {
+        private var savedInstanceStateExpected = false
+        const val RECOVERED_FROM_CRASH = "RECOVERED_FROM_CRASH"
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.kt:185]: Use (if possible private) compile-time constant instead, please.
Please check the other constants too.

**Review by gPathpp — CHANGES_REQUESTED** [review]
Nice 👍 Please refactor the constants

File: catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.kt (hunk 1 above)
Comment (khushal1707) [reply catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.kt:185]: In this case private is not possible as this const used by other file ( `BaseExceptionHandler` ), I tried internal modifier please have a look :)

File: catroid/src/main/java/org/catrobat/catroid/ui/BaseExceptionHandler.kt [hunk 2]
```diff
@@ -20,38 +20,34 @@
...
+    }
+
+    protected fun exit() {
+        Process.killProcess(Process.myPid())
+        exitProcess(EXIT_CODE)
+    }
+
+    companion object {
+        private val TAG = BaseExceptionHandler::class.java.simpleName
+        private const val EXIT_CODE = 10
```
Comment (gPathpp) [inline catroid/src/main/java/org/catrobat/catroid/ui/BaseExceptionHandler.kt:51]: Compile-time constant, please.

**Review by gPathpp — CHANGES_REQUESTED** [review]

**Comment by khushal1707** [comment]
Done!

**Review by gPathpp — APPROVED** [review]
LGTM

**Comment by ndrnour** [comment]
Thanks!

File: catroid/src/main/java/org/catrobat/catroid/ui/BaseExceptionHandler.kt [hunk 3]
```diff
@@ -20,38 +20,35 @@
...
+        preferences.edit()
+            .putBoolean(RECOVERED_FROM_CRASH, true)
+            .apply()
+        exit()
+    }
 
-	private final SharedPreferences preferences;
+    protected fun exit() {
+        Process.killProcess(Process.myPid())
+        exitProcess(EXIT_CODE)
```
Comment (ndrnour) [inline catroid/src/main/java/org/catrobat/catroid/ui/BaseExceptionHandler.kt:48]: 👍🏻

**Pull request merged**